src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -336,51 +385,79 @@ by { rw [← coe_fn_comp_to_finite_measure_eq_coe_fn,
-is `probability_measure.test_against_nn`. -/
-def test_against_nn
-  (μ : probability_measure α) (f : α →ᵇ ℝ≥0) : ℝ≥0 :=
-(lintegral (μ : measure α) ((coe : ℝ≥0 → ℝ≥0∞) ∘ f)).to_nnreal
-
 lemma lintegral_lt_top_of_bounded_continuous_to_nnreal (μ : probability_measure α) (f : α →ᵇ ℝ≥0) :
   ∫⁻ x, f x ∂(μ : measure α) < ∞ :=
 μ.to_finite_measure.lintegral_lt_top_of_bounded_continuous_to_nnreal f
 
-@[simp] lemma test_against_nn_coe_eq {μ : probability_measure α} {f : α →ᵇ ℝ≥0} :
-  (μ.test_against_nn f : ℝ≥0∞) = ∫⁻ x, f x ∂(μ : measure α) :=
-ennreal.coe_to_nnreal (lintegral_lt_top_of_bounded_continuous_to_nnreal μ f).ne
-
-@[simp] lemma to_finite_measure_test_against_nn_eq_test_against_nn
-  {μ : probability_measure α} {f : α →ᵇ nnreal} :
-  μ.to_finite_measure.test_against_nn f = μ.test_against_nn f := rfl
-
-lemma test_against_nn_const (μ : probability_measure α) (c : ℝ≥0) :
-  μ.test_against_nn (bounded_continuous_function.const α c) = c :=
-by simp [← ennreal.coe_eq_coe, (measure_theory.is_probability_measure μ).measure_univ]
-
-lemma test_against_nn_mono (μ : probability_measure α)
-  {f g : α →ᵇ ℝ≥0} (f_le_g : (f : α → ℝ≥0) ≤ g) :
-  μ.test_against_nn f ≤ μ.test_against_nn g :=
-by simpa using μ.to_finite_measure.test_against_nn_mono f_le_g
-
 variables [opens_measurable_space α]
 
 lemma test_against_nn_lipschitz (μ : probability_measure α) :
-  lipschitz_with 1 (λ (f : α →ᵇ ℝ≥0), μ.test_against_nn f) :=
+  lipschitz_with 1 (λ (f : α →ᵇ ℝ≥0), μ.to_finite_measure.test_against_nn f) :=
 begin
   have key := μ.to_finite_measure.test_against_nn_lipschitz,
   rwa μ.mass_to_finite_measure at key,
 end
 
+/-- The topology of weak convergence on `probability_measures α`. This is inherited (induced) from
+the weak-*  topology on `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)` via the function
+`probability_measures.to_weak_dual_bcnn`. -/
+instance : topological_space (probability_measure α) :=
+topological_space.induced (λ (μ : probability_measure α), μ.to_finite_measure) infer_instance
+
+lemma to_finite_measure_continuous :
+  continuous (to_finite_measure : probability_measure α → finite_measure α) :=
+continuous_induced_dom
+
 /-- Probability measures yield elements of the `weak_dual` of bounded continuous nonnegative
-functions via `probability_measure.test_against_nn`, i.e., integration. -/
-def to_weak_dual_bounded_continuous_nnreal (μ : probability_measure α) :
-  weak_dual ℝ≥0 (α →ᵇ ℝ≥0) :=
-{ to_fun := λ f, μ.test_against_nn f,
-  map_add' := μ.to_finite_measure.test_against_nn_add,
-  map_smul' := μ.to_finite_measure.test_against_nn_smul,
-  cont := μ.test_against_nn_lipschitz.continuous, }
+functions via `finite_measure.test_against_nn`, i.e., integration. -/
+def to_weak_dual_bcnn : probability_measure α → weak_dual ℝ≥0 (α →ᵇ ℝ≥0) :=
+finite_measure.to_weak_dual_bcnn ∘ to_finite_measure
+
+lemma to_weak_dual_bcnn_eval (μ : probability_measure α) (f : α →ᵇ ℝ≥0) :
+  μ.to_weak_dual_bcnn f = (∫⁻ x, f x ∂(μ : measure α)).to_nnreal := rfl
```

Comment:
I think you want this simp lemma again. Also, `eval` should be `apply`. I'm not sure you need the `apply` lemma though, given the definition in terms of `to_finite_measure`. Note, don't try to use `simps` here (you don't quite get the right lemma).
```suggestion

@[simp] lemma coe_to_weak_dual_bcnn (μ : probability_measure α) :
  ⇑μ.to_weak_dual_bcnn = μ.to_finite_measure.test_against_nn := rfl

lemma to_weak_dual_bcnn_apply (μ : probability_measure α) (f : α →ᵇ ℝ≥0) :
  μ.to_weak_dual_bcnn f = (∫⁻ x, f x ∂(μ : measure α)).to_nnreal := rfl
```